pkg/controller/controller_instance.go

```diff
@@ -1534,12 +1574,14 @@ func (c *controller) processUpdateServiceInstanceSuccess(instance *v1beta1.Servi
 
 // processUpdateServiceInstanceFailure handles the logging and updating of a
 // ServiceInstance that hit a terminal failure during update reconciliation.
-func (c *controller) processUpdateServiceInstanceFailure(instance *v1beta1.ServiceInstance, readyCond *v1beta1.ServiceInstanceCondition) error {
+func (c *controller) processUpdateServiceInstanceFailure(instance *v1beta1.ServiceInstance, readyCond, failedCond *v1beta1.ServiceInstanceCondition, updateObservedGeneration bool) error {
 	c.recorder.Event(instance, corev1.EventTypeWarning, readyCond.Reason, readyCond.Message)
 	currentReconciledGeneration := instance.Status.ReconciledGeneration
 
 	setServiceInstanceCondition(instance, v1beta1.ServiceInstanceConditionReady, readyCond.Status, readyCond.Reason, readyCond.Message)
-	clearServiceInstanceCurrentOperation(instance)
+	setServiceInstanceCondition(instance, v1beta1.ServiceInstanceConditionFailed, failedCond.Status, failedCond.Reason, failedCond.Message)
```

Comment:
**This is a bugfix**: once I switched from `ReconciledGeneration` to `ObservedGeneration` + `Failed:True`, I found that some integration tests were failing on waiting for this condition.
For provisioning we always set the `Failed:True` condition for terminal errors, and for updates we never were.
I think this is a bug, and fixed it for consistency.

Reply:
I'm definitely in favor of this, though we should make sure this keeps the current behavior of spec changes to a failed update triggering a new update.

From a cursory glance, it seems like that'd require 1) removing the L383 block on the `Failed` condition, and 2) removing the `Failed` condition when recording the operation start at L422.

Reply:
I don't think that it is a bug. The semantics of `Failed:True` are that the instance is not usable. If an update fails, the instance is still usable, in its previous state. With that being said, I am on board with changing the semantics of `Failed:True` to mean that the previous operation failed.

Reply:
That's true. It's not *really* a bug, more just an unfortunate design choice that can be easily rectified.

Reply:
@staebler @kibbles-n-bytes by "bug" I mean that this method is not doing what the comment says:
```go
 // processUpdateServiceInstanceFailure handles the logging and updating of a
 // ServiceInstance that hit a terminal failure during update reconciliation.
```
The "terminal failure" here means that Instance should be marked with `Failed:True`, but it is not.
So I wanted to temporarily "fix" this to make it work as it was poorly designed, and then separately fix it according to the new design.

Reply:
> 1) removing the L383 block on the Failed condition, and 2) removing the Failed condition when recording the operation start at L422.

@kibbles-n-bytes I belive the other PRs are already addressing this: #1751 and #1765